iceberg-rust/src/table/transaction/append.rs

```diff
@@ -127,70 +129,122 @@ pub(crate) fn select_manifest_partitioned(
     let mut file_count_all_entries = 0;
     for manifest_res in manifest_list_reader {
         let manifest = manifest_res?;
         // TODO: should this also account for existing_rows_count / existing_files_count?
         let row_count = manifest.added_rows_count;
         file_count_all_entries += manifest.added_files_count.unwrap_or(0) as usize;
 
-        let Some((selected_row_count, selected_manifest)) = &selected_state else {
-            selected_state = Some((row_count, manifest));
-            continue;
-        };
-
-        // If the file doesn't have any rows, we select it
-        let Some(row_count) = row_count else {
-            selected_state = Some((row_count, manifest));
-            continue;
-        };
-
-        if selected_row_count.is_some_and(|x| x > row_count) {
-            manifest_list_writer.append_ser(selected_manifest)?;
-            selected_state = Some((Some(row_count), manifest));
-            continue;
-        } else {
-            manifest_list_writer.append_ser(manifest)?;
-            continue;
+        match manifest.content {
+            iceberg_rust_spec::manifest_list::Content::Data => {
+                let Some((selected_row_count, selected_manifest)) = &selected_data_state else {
+                    selected_data_state = Some((row_count, manifest));
+                    continue;
+                };
+
+                // If the file doesn't have any rows, we select it
+                let Some(row_count) = row_count else {
+                    selected_data_state = Some((row_count, manifest));
+                    continue;
+                };
+
+                if selected_row_count.is_some_and(|x| x > row_count) {
+                    manifest_list_writer.append_ser(selected_manifest)?;
+                    selected_data_state = Some((Some(row_count), manifest));
+                    continue;
+                } else {
+                    manifest_list_writer.append_ser(manifest)?;
+                    continue;
+                }
+            }
+            iceberg_rust_spec::manifest_list::Content::Deletes => {
+                let Some((selected_row_count, selected_manifest)) = &selected_delete_state else {
+                    selected_delete_state = Some((row_count, manifest));
+                    continue;
+                };
+
+                // If the file doesn't have any rows, we select it
+                let Some(row_count) = row_count else {
+                    selected_delete_state = Some((row_count, manifest));
+                    continue;
+                };
+
+                if selected_row_count.is_some_and(|x| x > row_count) {
+                    manifest_list_writer.append_ser(selected_manifest)?;
+                    selected_delete_state = Some((Some(row_count), manifest));
+                    continue;
+                } else {
+                    manifest_list_writer.append_ser(manifest)?;
+                    continue;
```

Comment:
These `continue`s are redundant; likewise above (lines 143, 147, 161, 165, 211, 214).

Besides that I find the let-some-else-continue less readable so i refactored those in my pr, though that's not a big deal.

Reply:
I'll have look, thanks. I wanted to quickly sketch out the general idea. Haven't looked into that yet.